### ISSUE TITLE
Create Deep Learning Network Architecture

### ISSUE THREAD

**Issue by shrutikadam3003** [shrutikadam3003/shrutikadam3003#3]
#Create Deep Learning Network Architecture
Script for creating the layers for a deep learning network with the following properties:
_**Number of layers: 70
Number of connections: 77**_
Run the script to create the layers in the workspace variable net.
To learn more, see Generate MATLAB Code From Deep Network Designer.
Auto-generated by MATLAB on 25-Apr-2025 22:00:19
Create dlnetwork
Create the dlnetwork variable to contain the network layers.
### net = dlnetwork;

**Add Layer Branches**
Add branches to the dlnetwork. Each branch is a linear array of layers.
.......................................................................................................................................................................................................................................

tempNet = [
    imageInputLayer([224 224 3],"Name","data","Normalization","zscore")
    convolution2dLayer([7 7],64,"Name","conv1","BiasLearnRateFactor",0,"Padding",[3 3 3 3],"Stride",[2 2])
    batchNormalizationLayer("Name","bn_conv1")
    reluLayer("Name","conv1_relu")
    maxPooling2dLayer([3 3],"Name","pool1","Padding",[1 1 1 1],"Stride",[2 2])];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],64,"Name","res2a_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn2a_branch2a")
    reluLayer("Name","res2a_branch2a_relu")
    convolution2dLayer([3 3],64,"Name","res2a_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn2a_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res2a")
    reluLayer("Name","res2a_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],64,"Name","res2b_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn2b_branch2a")
    reluLayer("Name","res2b_branch2a_relu")
    convolution2dLayer([3 3],64,"Name","res2b_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn2b_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res2b")
    reluLayer("Name","res2b_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],128,"Name","res3a_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1],"Stride",[2 2])
    batchNormalizationLayer("Name","bn3a_branch2a")
    reluLayer("Name","res3a_branch2a_relu")
    convolution2dLayer([3 3],128,"Name","res3a_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn3a_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([1 1],128,"Name","res3a_branch1","BiasLearnRateFactor",0,"Stride",[2 2])
    batchNormalizationLayer("Name","bn3a_branch1")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res3a")
    reluLayer("Name","res3a_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],128,"Name","res3b_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn3b_branch2a")
    reluLayer("Name","res3b_branch2a_relu")
    convolution2dLayer([3 3],128,"Name","res3b_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn3b_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res3b")
    reluLayer("Name","res3b_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],256,"Name","res4a_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1],"Stride",[2 2])
    batchNormalizationLayer("Name","bn4a_branch2a")
    reluLayer("Name","res4a_branch2a_relu")
    convolution2dLayer([3 3],256,"Name","res4a_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn4a_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([1 1],256,"Name","res4a_branch1","BiasLearnRateFactor",0,"Stride",[2 2])
    batchNormalizationLayer("Name","bn4a_branch1")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res4a")
    reluLayer("Name","res4a_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],256,"Name","res4b_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn4b_branch2a")
    reluLayer("Name","res4b_branch2a_relu")
    convolution2dLayer([3 3],256,"Name","res4b_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn4b_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res4b")
    reluLayer("Name","res4b_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],512,"Name","res5a_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1],"Stride",[2 2])
    batchNormalizationLayer("Name","bn5a_branch2a")
    reluLayer("Name","res5a_branch2a_relu")
    convolution2dLayer([3 3],512,"Name","res5a_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn5a_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([1 1],512,"Name","res5a_branch1","BiasLearnRateFactor",0,"Stride",[2 2])
    batchNormalizationLayer("Name","bn5a_branch1")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res5a")
    reluLayer("Name","res5a_relu")];
net = addLayers(net,tempNet);

tempNet = [
    convolution2dLayer([3 3],512,"Name","res5b_branch2a","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn5b_branch2a")
    reluLayer("Name","res5b_branch2a_relu")
    convolution2dLayer([3 3],512,"Name","res5b_branch2b","BiasLearnRateFactor",0,"Padding",[1 1 1 1])
    batchNormalizationLayer("Name","bn5b_branch2b")];
net = addLayers(net,tempNet);

tempNet = [
    additionLayer(2,"Name","res5b")
    reluLayer("Name","res5b_relu")
    globalAveragePooling2dLayer("Name","pool5")
    fullyConnectedLayer(1000,"Name","fc1000")
    softmaxLayer("Name","prob")];
net = addLayers(net,tempNet);

% clean up helper variable
clear tempNet;
.......................................................................................................................................................................................................................................
Connect Layer Branches
Connect all the branches of the network to create the network graph.
net = connectLayers(net,"pool1","res2a_branch2a");
net = connectLayers(net,"pool1","res2a/in2");
net = connectLayers(net,"bn2a_branch2b","res2a/in1");
net = connectLayers(net,"res2a_relu","res2b_branch2a");
net = connectLayers(net,"res2a_relu","res2b/in2");
net = connectLayers(net,"bn2b_branch2b","res2b/in1");
net = connectLayers(net,"res2b_relu","res3a_branch2a");
net = connectLayers(net,"res2b_relu","res3a_branch1");
net = connectLayers(net,"bn3a_branch2b","res3a/in1");
net = connectLayers(net,"bn3a_branch1","res3a/in2");
net = connectLayers(net,"res3a_relu","res3b_branch2a");
net = connectLayers(net,"res3a_relu","res3b/in2");
net = connectLayers(net,"bn3b_branch2b","res3b/in1");
net = connectLayers(net,"res3b_relu","res4a_branch2a");
net = connectLayers(net,"res3b_relu","res4a_branch1");
net = connectLayers(net,"bn4a_branch2b","res4a/in1");
net = connectLayers(net,"bn4a_branch1","res4a/in2");
net = connectLayers(net,"res4a_relu","res4b_branch2a");
net = connectLayers(net,"res4a_relu","res4b/in2");
net = connectLayers(net,"bn4b_branch2b","res4b/in1");
net = connectLayers(net,"res4b_relu","res5a_branch2a");
net = connectLayers(net,"res4b_relu","res5a_branch1");
net = connectLayers(net,"bn5a_branch2b","res5a/in1");
net = connectLayers(net,"bn5a_branch1","res5a/in2");
net = connectLayers(net,"res5a_relu","res5b_branch2a");
net = connectLayers(net,"res5a_relu","res5b/in2");
net = connectLayers(net,"bn5b_branch2b","res5b/in1");
net = initialize(net);
_________________________________________________________________________________________________________________________
### Plot Layers
plot(net);

xlim([0.840 2.160])
ylim([-9.1 74.1])
